### DISCUSSION TITLE
Add Modscanner support to Chatterino

### DISCUSSION THREAD

**Discussion by Walledgarden** [Chatterino/chatterino2#4885] [Ideas] [answered]
ModScanner is a platform, which offers users the ability to query and retrieve detailed information regarding any user on Twitch. The page provides users with insights into the selected channels, where the selected user is currently moderating, VIP, and founder. Also, ModScanner displays the users relating to a specified channels like the moderators, VIPs, and founders for the channel-sight as well.

Including a button at the profile card in Chatterino that redirects users to the ModScanner page can provide huge advantages. By using this feature, moderators can enhance their ability or normal users can learn more about a specific user to fulfill their job by gaining in-depth insights into a given user's involvement and roles within the Twitch community.

In the past, there was already another Lookup for Channel Moderators, which provided very good utility to the Twitch users, but was sadly shutdown recently, so we decided to make up our own page.

This is an example of the User Lookup page on https://modscanner.com/user/cupofeggy (or https://mod.sc/cupofeggy)
![image](https://github.com/Chatterino/chatterino2/assets/85127985/a60e14e8-ec2d-440c-84b1-96893baad676)

**Comment by iProdigy** [comment]
This should probably be left for a third-party plugin, rather than native to every c2 install. Chatterino didn't natively integrate the old ModLookup service & ModScanner provides incomplete coverage of mod/vip/etc data

**Comment by Felanbird** [comment, marked as answer]
This website uses Twitch's first-party API to obtain this data, which is not something we want to actively use in Chatterino.

The only way this feature would get added to Chatterino is if a collaborator wanted to build an opt-in system for streamers to expose this data, or as prodigy mentioned, via a plugin.

**Comment by Mm2PL** [comment]
I understand that you may not be happy that we'd like for this to be a plugin when plugin support isn't even finished yet. I'm slowly cooking up the necessary things to make developing plugins decent. Another issue is, as felanbird noted, that they use GQL to acquire the data. This website might get shut down if we show Twitch that it exists.
